gridtools/make_topog/topogobj.py

```diff
@@ -0,0 +1,88 @@
+from typing import List
+import xarray as xr
+import numpy as np
+import numpy.typing as npt
+import dataclasses
+
+# represents topography output file created by make_topog
+# contains parameters for topography generation that aren't tied to a specific topography type
+# and depth values from specified topog_type algorithm once generated.
+# if multiple tiles are used, the third index of depth will be the the tile number
+@dataclasses.dataclass
+class TopogObj():
+    output_name: str = None
+    ntiles: int = None
+    nx: int = None
+    ny: int = None
+    x_refine: int = None
+    y_refine: int = None
+    scale_factor: float = None
+    depth: npt.NDArray[np.float64] = None
+
+    ds: xr.Dataset = None
+    data_is_generated: bool = None
```

Comment:
will this be used outside of TopogObj?   Make it a "private" variable _data_is_generated? and initialize it as False here?